cluster-provision/k8s/1.24/provision.sh

```diff
@@ -99,6 +99,8 @@ enabled=1
 EOF
 dnf install -y cri-o
 
+systemctl enable --now crio
```

Comment:
Why do we enable crio at two places, this and [that one](https://github.com/kubevirt/kubevirtci/pull/800/commits/809e0476f9061a2dc510553436f25b197efc9606#diff-e27be89aa116a14f88ff53f83b384465a6b1d8af5c9e5615b75e136266544b51R42)?

Reply:
Note the 2nd one is needed after deleting the CNI folder
https://github.com/kubevirt/kubevirtci/pull/800/commits/809e0476f9061a2dc510553436f25b197efc9606#diff-e27be89aa116a14f88ff53f83b384465a6b1d8af5c9e5615b75e136266544b51R37

(not sure about the first one indeed)

Reply:
For the first one - cri-o needs to be running in order to pull the kubernetes images [here](https://github.com/kubevirt/kubevirtci/pull/800/files/f69b854f5981e792436640d5d2d25eb4db3c85c7..809e0476f9061a2dc510553436f25b197efc9606#diff-531d46398155ece227284b4b8c0bed55f400f96da4e41289f5dbec168cc607dbR142)

Reply:
I just found it strange. I think that duplication in general is a code smell that points towards some common functionality. OTOH as this is a very small occurence which I can live with.

IIRC "someone" (citation needed tm) once said, avoid premature generalization, before you've had enough cases to see the commonalities clearly.

Thanks for the explanation!

Reply:
To be more precise: there's a whole chapter about reuse in [Facts And Fallacies of Software Engineering](https://www.goodreads.com/book/show/83792.Facts_and_Fallacies_of_Software_Engineering) about it, very much worth reading :)